### PR TITLE
feat(pet-164): petasos.selfmod.* self-tamper detection family

### PR DESCRIPTION
**Pull request by ziomancer**
## Summary
- Add `petasos.selfmod.*` finding family: detection-only classification of agent tool calls targeting Petasos-owned config surfaces (config.yaml, profile homes, enforcement spool, console token API)
- Side-effect-free on allow/deny (PET-125 intact); escalation via floor-protected frequency weights (10.0/3.0) through the ordinary tier channel
- Dedicated cooldown-exempt alert delivery, unsuppressible by any config toggle except the documented master switches

## Two-layer detection

| Rule id | Trigger | Severity | Weight floor |
|---------|---------|----------|-------------|
| `petasos.selfmod.config_write` | Non-read-only tool, top-level arg value *equals* an owned path | CRITICAL | 10.0 |
| `petasos.selfmod.config_ref` | Non-read-only tool, owned path *contained* in any arg part | HIGH | 3.0 |
| `petasos.selfmod.console_probe` | Console `/api/*` 401 (token armed) | CRITICAL | n/a |

## Files changed

| File | Change |
|------|--------|
| `petasos/session/guard.py` | `READ_ONLY_TOOLS` frozenset (promoted); `_classify_selfmod()` detector; `GuardResult.selfmod_target/selfmod_finding` fields; post-verdict frequency update |
| `petasos/pipeline.py` | `record_selfmod()` (alert + audit only); `_is_floor_rule()` gains selfmod prefix |
| `petasos/session/frequency.py` | `DEFAULT_FREQUENCY_WEIGHTS` entries + inline floor literals in `_match_weight()` |
| `petasos/session/alerting.py` | `evaluate_selfmod()` delivery path (per-rule_id critical cap) |
| `petasos/session/profiles/__init__.py` | `_validate_suppress_rules()` strips `petasos.selfmod.*` |
| `docs/deployment/reference_plugin/__init__.py` | Import `READ_ONLY_TOOLS` from guard; emit spool `selfmod_attempt` |
| `petasos/console/_paths.py` | `display_path()`, `spool_path()`, `spool_rot_path()` hoisted |
| `petasos/console/_events.py` | Delegates spool override to `_paths` |
| `petasos/console/server.py` | `_maybe_fire_console_probe()` on 401 seam |
| `docs/deployment/hardening.md` | Self-tamper detection limits paragraph |
| `tests/conftest.py` | Retarget `_SPOOL_PATH_OVERRIDE` to `_paths` |
| `tests/test_selfmod_detection.py` | New: 27 behavioral tests |
| `tests/test_selfmod_floor_shape.py` | New: 10 static/AST shape tests |
| `tests/adversarial/selfmod/test_selfmod_suppression_bypass.py` | New: adversarial suppression bypass suite |

## Test plan

- [x] `C:\python310\python.exe -m pytest tests/ -q --deselect tests/test_console_validation.py::test_default_ignorable_rejected` -- 2211/2211 pass (full output: docs/specs/TODO/PET-164.test-output.txt)
- [x] Shape/static tests pin all five floor invariants (Decision 12)
- [x] Adversarial tests: weight zeroing, alert_enabled: false, frequency_enabled: false, inflated thresholds, profile suppress_rules, severity_overrides, all built-in profiles

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added detection-only self-tamper classification for tool calls targeting protected configuration paths, including critical/high severity, rate-limited alerting, and frequency “floor” escalation.
  * Added console access probing on failed authentication attempts (does not affect access outcomes).
  * Added enforcement spool path override support and improved user-friendly path display.
* **Documentation**
  * Documented self-tamper behavior, operator tuning knobs, known detection misses/tripwires, and false-positive guidance.
* **UI Updates**
  * Updated console event rendering for self-tamper attempts with an amber badge and dedicated “Self-tamper attempt” detection-only details.
* **Tests**
  * Expanded self-tamper, suppression, frequency-weight, and console UI rendering coverage (including edge cases).

### CODE REVIEW

File: petasos/console/server.py [hunk 1]
```diff
@@ -61,6 +61,10 @@
 _MAX_TALLY_SESSIONS = 10_000
 # Poll cadence of the dashboard's background enforcement-spool tailer (standalone).
 _ENFORCEMENT_TAIL_INTERVAL_S = 1.0
+# PET-164 (D9): rate-limit for console_probe selfmod on 401. One recording per
+# interval so a brute-force loop cannot spam the audit/alert path.
+_selfmod_401_last: float = 0.0
+_SELFMOD_401_INTERVAL: float = 10.0
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/server.py:66]: Resolved in 2461868

File: petasos/session/guard.py [hunk 2]
```diff
@@ -186,6 +223,113 @@ def __init__(
...
+            top_level_strs: list[str] = []
+            all_parts: list[str] = []
+            for value in tool_params.values():
+                if value is None:
+                    continue
+                if isinstance(value, str):
+                    top_level_strs.append(value)
+                    all_parts.append(value)
+                else:
+                    all_parts.append(safe_json_dumps(value))
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/session/guard.py]: Resolved in 2461868

File: petasos/session/guard.py [hunk 3]
```diff
@@ -186,6 +223,113 @@ def __init__(
...
+                s = s.replace("\\\\", "/").replace("\\", "/").replace("/", os.sep)
+                return os.path.normcase(s)
+
+            for tls in top_level_strs:
+                normed = _normalize_for_compare(tls)
+                for entry in owned:
+                    if normed == entry:
+                        return _SelfmodMatch(
+                            rule_id="petasos.selfmod.config_write", target=entry
+                        )
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/session/guard.py]: Resolved in 2461868

File: tests/test_selfmod_detection.py [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+from unittest.mock import patch
+
+import pytest
+
+from petasos._types import ScanFinding, Severity
+from petasos.config import PetasosConfig
+from petasos.pipeline import Pipeline
+from petasos.session.frequency import FrequencyTracker
+from petasos.session.guard import READ_ONLY_TOOLS, GuardResult, ToolCallGuard, _READ_ONLY_CANON
+from petasos.session.profiles import ResolvedProfile, TierThresholds
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply tests/test_selfmod_detection.py]: Resolved in 2461868

File: tests/test_selfmod_detection.py [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    ) -> None:
+        pipeline, guard, tracker, owned = guard_pair
+        result = await guard.evaluate(
+            "write_file", {"path": "safe.txt"}, "s1"
+        )
+        d = result.to_dict()
+        assert d["selfmod_target"] is None
+        assert d["selfmod_finding"] is None
+
+    async def test_to_dict_selfmod_finding_is_dict_when_present(
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply tests/test_selfmod_detection.py:400]: Resolved in 2461868 -- superset of finding 4, all 3 test modules fixed.

**Comment by ziomancer** [comment]
@coderabbitai full review

File: docs/deployment/hardening.md [hunk 6]
```diff
@@ -333,6 +333,39 @@ short-circuit `_pre_tool_call` before any scan runs, so a disarmed pipeline neve
...
+  surfaces and fire regardless -- at least 1/min per class under a hostile cap config.
+
+`alert_enabled` and `frequency_enabled` are **not** off-switches for this class:
+`record_selfmod`'s alert bypasses the alerting gate by construction, and the
+frequency update runs in the guard on its own tracker.
+
+**Accepted detection misses (tripwire, not wall):** relative-path tricks,
+symlinked aliases, `$HERMES_HOME` indirection inside shell strings, ancestor-
+directory rename/delete, and owned paths beyond `safe_json_dumps`' depth/size
+caps. The OS boundary remains the containment mechanism; this layer is a
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply docs/deployment/hardening.md]: Resolved in 1d9b829

File: petasos/session/guard.py [hunk 7]
```diff
@@ -186,6 +224,132 @@ def __init__(
...
+            out: list[str] = []
+            for v in value.values():
+                out.extend(ToolCallGuard._extract_leaf_strings(v, depth + 1))
+            return out
+        if isinstance(value, (list, tuple)):
+            out = []
+            for item in value:
+                out.extend(ToolCallGuard._extract_leaf_strings(item, depth + 1))
+            return out
+        return []
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/session/guard.py:304]: Resolved in 1d9b829

File: tests/test_selfmod_floor_shape.py [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    """Pin 3: evaluate_selfmod contains 'critical' and no _rule_cooldowns access."""
+
+    def test_contains_critical_literal(self) -> None:
+        func = _find_method_in_class(
+            _SRC / "session" / "alerting.py",
+            "AlertManager",
+            "evaluate_selfmod",
+        )
+        source = ast.dump(func)
+        assert "'critical'" in source or '"critical"' in source or "critical" in source
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply tests/test_selfmod_floor_shape.py]: Resolved in 1d9b829

**Comment by ziomancer** [comment]
Round 3: live-fire review fixes (empirical harness against the real guard).

**What was found and fixed:**
1. **Depth-overflow FP message** - args nested past the depth-32 cap flagged as self-tamper naming an *arbitrary owned path* as the target, even with no owned path anywhere in the args. Now records the dedicated `<argument-depth-overflow>` marker and an honest 'no owned path matched' message. 3 new behavioral tests.
2. **Console rendering gap** - `selfmod_attempt` spool rows rendered with a green *safe* badge and the drill-down fell into the *Unknown row kind* fallback (rule/severity/reason stored but never shown). Now: amber **self-tamper** badge, dedicated drill-down branch with detection-only explainer, honest 'scan ran: n/a' provenance. New `tests/js/selfmod-row.test.mjs` (5 tests).
3. **hardening.md** - documents the overflow marker + operator note that legitimate rapid config edits escalate like an attack (tier2 at the 3rd rapid write under defaults), with unequip-or-space guidance.

**Verified behaviors (no code change needed):** 25-call realistic coding stream produced zero selfmod findings (~6us/call classifier overhead); disarm contract holds (unequipped and `tool_guard_enabled: false` fully silence the class; `alert_enabled`/`frequency_enabled`/`audit_enabled` correctly do not).

Suites: 2214 passed / 1 known deselect (pytest), 241/241 (node --test). Richer console surfacing (dedicated tile/filter) deferred to a follow-up ticket.

File: docs/deployment/hardening.md [hunk 9]
```diff
@@ -333,6 +333,52 @@ short-circuit `_pre_tool_call` before any scan runs, so a disarmed pipeline neve
...
+direction; widen `READ_ONLY_TOOLS` as needed).
+
+**Operator note (legitimate config edits escalate).** The frequency weights do
+not distinguish an operator-directed edit session from an attack: rapid
+repeated writes to the owned config reach tier 1 at the second write, tier 2
+(all tool calls blocked) at the third, and tier 3 at the fifth under default
+thresholds, and shell reads (`cat`, `tail`) of the config or spool weigh 3.0
+each. When you direct an agent to edit the `petasos:` section, either flip the
+Equipped switch off for the duration (the documented master off-switch) or
+space the writes beyond the 60 second frequency half-life.
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply docs/deployment/hardening.md]: Resolved in f485788

File: petasos/console/static/petasos.js [hunk 10]
```diff
@@ -1246,10 +1261,14 @@
...
+      // PET-164: a self-tamper classification row is detection-only (safe=true on the
+      // summary), but it must never wear the green "safe" badge; amber "self-tamper"
+      // keeps red reserved for actual blocks.
+      var isSelfmodRow = isEnforcement && et === "selfmod_attempt";
       var isBlocked = (e.safe === false); // strict ===false; truthy-but-not-false is not "blocked"
 
-      var badgeText = isBypass ? "bypassed (disarmed)" : (isBlocked ? "blocked" : "safe");
-      var badgeClass = isBypass ? "warn" : (isBlocked ? "err" : "ok");
+      var badgeText = isBypass ? "bypassed (disarmed)" : (isSelfmodRow ? "self-tamper" : (isBlocked ? "blocked" : "safe"));
+      var badgeClass = (isBypass || isSelfmodRow) ? "warn" : (isBlocked ? "err" : "ok");
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/static/petasos.js:1271]: Resolved in f485788

File: petasos/console/static/petasos.js [hunk 11]
```diff
@@ -1143,6 +1146,18 @@
...
+      // never changed this call's allow/deny; repeated attempts escalate the session
+      // tier and a dedicated alert fires in the gateway log.
+      wrap.appendChild(Pet.h("div", { className: "sd-heartbeat" },
+        Pet.h("div", {}, "Self-tamper attempt: this call referenced a Petasos-owned config surface."),
+        Pet.h("div", { className: "sd-sub" }, "Detection only: the call itself was not blocked by this classification. Repeated attempts escalate the session tier; a dedicated alert fires in the gateway log.")));
+      wrap.appendChild(fld("tool", e.tool));
+      wrap.appendChild(fld("rule", e.rule_id));
+      wrap.appendChild(fld("severity", e.severity));
+      wrap.appendChild(fld("reason", e.reason));
+      wrap.appendChild(fld("session", e.session_id));
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/static/petasos.js:1160]: Skipped — text is accurate; record_selfmod bypasses the alerting gate by construction (see hardening.md "alert_enabled and frequency_enabled are not off-switches for this class"). The alert fires whenever selfmod is detected, which itself requires the system to be enabled.

File: petasos/console/server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/console/server.py:66]: Already addressed in a prior commit — _selfmod_401_last is now dict[int, float] keyed by id(pipeline), scoping the rate-limit per console app instance.

File: petasos/session/guard.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/session/guard.py]: Already addressed in a prior commit — _normalize_for_compare now calls Path.resolve(strict=False) for absolute paths before comparison.

File: tests/test_selfmod_detection.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ziomancer) [reply tests/test_selfmod_detection.py]: Already addressed in a prior commit — all lint and typecheck CI lanes pass.

File: tests/test_selfmod_detection.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ziomancer) [reply tests/test_selfmod_detection.py:400]: Already addressed in a prior commit — all three test modules pass lint and strict-mypy CI gates.
